### PR TITLE
Separating user types and features for each user type in user-page (types: charity, donor, unset)

### PR DESCRIPTION
**Pull request by pkhodadoustan**

### CODE REVIEW

**Comment by pkhodadoustan** [comment]
This feature is not complete yet, but I have break it to several steps. Step one is done, which is separating (assigning) user types: "charity", "donor", "unset". The difficulty of this task comes from the fact that "type" attribute added to the User class is new! previously added users stored in data store do not have this attribute. Thus, this code should take care of those users by giving them a default "unset" type, same as new users before they set their type to either charity or donor. I suggest you do not use "mvn clean" before testing this feature (it's better to have some previously saved users and data) so that we can be sure that the code would not break when handling previously added users who did not have the attribute "type".

File: src/main/java/com/google/codeu/data/User.java [hunk 1]
```diff
@@ -2,12 +2,21 @@
...
-    public User(String email, String aboutMe) {
+    private Long type;
+    public User(String email, String aboutMe, long type) {
         this.email = email;
         this.aboutMe = aboutMe;
+        this.type = type;
+    }
+    
+    public User(String email, String aboutMe) {
+        this(email, aboutMe, -1);
```
Comment (ginellegaisano) [inline src/main/java/com/google/codeu/data/User.java]: nit: should use UNSET_TYPE directly.

**Review by ginellegaisano — APPROVED** [review]
It's working from my end :). Just a nit.

File: src/main/java/com/google/codeu/data/User.java (hunk 1 above)
Comment (pkhodadoustan) [reply src/main/java/com/google/codeu/data/User.java]: Thank you, I'll fix it

**Pull request merged**